geograypher/meshes/meshes.py

```diff
@@ -1320,12 +1328,21 @@ def pix2face(
             pix2face_list = [
                 self.pix2face(camera, render_img_scale=render_img_scale)
                 for camera in cameras
-            ]
+            ]            
             pix2face = np.stack(pix2face_list, axis=0)
             return pix2face
-
+        
         ## Single camera case
 
+        # Check if the cachce contains a valid pix2face for the camera based on the hash
+        mesh_hash = self.get_mesh_hash()
+        camera_hash = cameras.get_camera_hash()
+        cacher = ub.Cacher('pix2face', depends=[mesh_hash, camera_hash, render_img_scale])
```

Comment:
It would be nice to explain the `on_error` since we were initially confused about it.